### PR TITLE
feat: improve settings UI

### PR DESCRIPTION
**Pull request by ReidyT**
see issue #72.

Here you can see directly a before/after comparaison:

(Before)
image

(After)
image

### CODE REVIEW

File: src/components/common/settings/SwitchModes.tsx [hunk 1]
```diff
@@ -11,7 +11,11 @@ import { USE_CHATBOT_DATA_CY } from '../../../config/selectors';
 import { DEFAULT_MARGIN } from '../../../config/stylingConstants';
 import { useAppSettingContext } from '../../context/AppSettingContext';
 
-const SwitchModes: FC = () => {
+type Prop = {
+  onChange?: (useChatbot: boolean) => void;
```
Comment (spaenleh) [inline src/components/common/settings/SwitchModes.tsx:15]: Why is the callback optional ?

**Review by spaenleh — COMMENTED** [review]
A general comment on naming:
- boolean: prefer `isSomething`
- callbacks: classic names include `onClick`, `onChange`, `onUpdate` ... always singular
- values: singular when refering to an object, plural when passing an array, i.e: `config` would be `{setting1: "value", setting2: "value2"}` and `configs` would be `[{toto: "1"}, {toto: "2"}]`

I am not sure to understand the purpose and issue that the `ObserverContext` is solving, see my comment. We can discuss this offline too.

File: src/components/views/admin/BuilderView.tsx [hunk 2]
```diff
@@ -38,32 +71,102 @@ const BuilderView: FC = () => {
...
+        sx={{ margin: DEFAULT_MARGIN }}
+      >
+        <GraaspButton
+          buttonDataCy={SETTINGS_SAVE_BUTTON_CY}
+          handleOnClick={handleButtonClicked}
+          sx={{
+            xs: { margin: DEFAULT_MARGIN },
+            sm: { margin: DEFAULT_MARGIN },
+          }}
+          minHeight="55px"
```
[the remark this answers is not in the document]
Comment (ReidyT) [reply src/components/views/admin/BuilderView.tsx:267]: It was already set in the code. To have the save button the same height as the add button, I didn't change this.

File: src/components/views/admin/BuilderView.tsx [hunk 3]
```diff
@@ -38,32 +71,102 @@ const BuilderView: FC = () => {
...
+          disabled={saveButtonDisabled}
+          text="Save"
```
[the remark this answers is not in the document]
Comment (ReidyT) [reply src/components/views/admin/BuilderView.tsx:269]: It will be in another PR. I'll open a issue for that.

**Review by spaenleh — APPROVED** [review]

**Review by spaenleh — APPROVED** [review]

**Pull request merged**